src/Symfony/Component/DomCrawler/FormFieldRegistry.php

```diff
@@ -124,13 +124,15 @@ public function has($name)
     public function set($name, $value)
     {
         $target = &$this->get($name);
-        if (!is_array($value) || $target instanceof Field\ChoiceFormField) {
+        if ($target instanceof Field\FormField) {
```

Comment:
@xabbuh there was an assumption that if the value is not an array, the `$target` is always an object. `ChoiceFormField` is the only implementation of `FormField` that works with arrays, therefore the special condition was there. `FormFieldRegistry::create()` expects a string as its first argument, and it would never worked with any of the `FormField` classes.

I think how conditions are currently organised makes it easier to understand.